api/v1beta1/trex_performance_types.go

```diff
@@ -0,0 +1,248 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Hash - struct to add hashes to status
+/*
+type Hash struct {
+	// Name of hash referencing the parameter
+	Name string `json:"name,omitempty"`
+	// Hash
+	Hash string `json:"hash,omitempty"`
+}
+*/
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// TrexPerformanceSpec defines the desired state of TrexPerformance
+type TrexPerformanceSpec struct {
```

Comment:
Just a thinking ... Shouldn't we choose other name than TrexPerformance? Maybe `AnsibleTests`?

Is the goal to have generic ansible tests runner, right?:)

Reply:
@eshulman2 I'm just wondering. Do we still want to go in the direction of the generic `AnsibleTests` CR or you encountered some issues that do not make it possible?

Reply:
done